### PR TITLE
Added ability to include CA certificates in the built image

### PR DESCRIPTION
**Pull request by jdob**
Misc other fixes:
* Reordered directory listing in building-images to be alphabetical
* Removed blank line and corrected log message capitalization in elemental.go
* Fixed typo in files_io.go

### CODE REVIEW

File: pkg/combustion/elemental.go [hunk 1]
```diff
@@ -24,10 +24,9 @@ const (
 var elementalScript string
 
 func configureElemental(ctx *image.Context) ([]string, error) {
-
 	if !isComponentConfigured(ctx, elementalConfigDir) {
 		log.AuditComponentSkipped(elementalComponentName)
-		zap.S().Info("Skipping elemental registration component, configuration is not provided")
+		zap.S().Info("skipping elemental registration component, configuration is not provided")
```
Comment (atanasdinov) [inline pkg/combustion/elemental.go:29]: I'm pretty sure 95%+ of the log messages are capitalised. Do we want to use lower cased ones instead?

File: pkg/combustion/certificates.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"github.com/suse-edge/edge-image-builder/pkg/log"
+	"github.com/suse-edge/edge-image-builder/pkg/template"
+	"go.uber.org/zap"
+)
+
+const (
+	certsComponentName = "certificates"
+	certsScriptName    = "30-certificates.sh"
```
Comment (atanasdinov) [inline pkg/combustion/certificates.go:18]: This seems rather low on the priority list. The task itself mentions:
> but likely needs to do it before Elemental/SUMA/Proxy is used.

I'm also wondering if this component should also differentiate the usages e.g. we'd eventually want to be able to provide custom certificates for Kubernetes clusters ([RKE2 docs for reference](https://docs.rke2.io/security/certificates#using-custom-ca-certificates)). Is this something we'd track separately? @rdoxenham

Comment (rdoxenham) [reply pkg/combustion/certificates.go:18]: I agree, this should be moved to a much higher priority in my opinion. If the proxy has a self-signed certificate (or they want to make sure it's actually the right proxy server being used), or the same situation for SUMA/Elemental/etc. then we need to make sure that the certificates are imported much earlier in the process. Suggest we go with 07 so it runs before the proxy is setup, especially as the proxy may be required for *other* things.

File: docs/building-images.md [hunk 3]
```diff
@@ -144,14 +144,16 @@ the structure of this directory will be better fleshed out. For now, the require
 
 There are a number of optional directories that may be included in the image configuration directory:
 
-* `network` - If present, network configurations will be generated from all desired states in this directory
-  and will be included in the built image. The configurations relevant for the particular host will be identified
-  and applied during the combustion phase. Check [nm-configurator](https://github.com/suse-edge/nm-configurator/)
-  for more information.
+* `certificates` - If present, all files with the extension ".pem" will be installed as CA certificates
+  in the built image.
```
Comment (rdoxenham) [inline docs/building-images.md:148]:
```suggestion
* `certificates` - If present, all files with the extension ".pem" will be installed as CA certificates
  in the built image. Note that ".crt" files are permitted, but please ensure they have a ".pem" extension.
```

File: pkg/combustion/templates/30-certificates.sh.tpl [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+set -euo pipefail
+
+cp ./{{ .CertificatesDir }}/* /etc/pki/trust/anchors/.
+update-ca-certificates
```
Comment (rdoxenham) [inline pkg/combustion/templates/30-certificates.sh.tpl:5]: We may want to think about making this output verbose so users can see this happening during boot up logs.

```suggestion
update-ca-certificates -v
```

**Review by rdoxenham — CHANGES_REQUESTED** [review]
We may want to think about ".crt" files as well... these should work as `update-ca-certificates` should process these accordingly. Right now the code explicitly looks for ".pem" files so I've added a suggestion for the documentation, but perhaps we also allow the user to specify ".crt" too? Also a note that we should make the certificates happen much earlier on in the boot process. Thanks!

File: pkg/combustion/certificates.go (hunk 2 above)
Comment (rdoxenham) [reply pkg/combustion/certificates.go:18]: @atanasdinov yes, we would definitely want to provide a way of specifying Kubernetes CA certificates. Maybe a subdirectory within the `cerfificates/` directory (or perhaps one inside of `kubernetes/`?) would be great to have at a later date. We should track this separately.

Comment (atanasdinov) [reply pkg/combustion/certificates.go:18]: Probably it makes more sense for those to be under `kubernetes/` then. Thanks for the clarification.

File: pkg/combustion/elemental.go (hunk 1 above)
Comment (jdob) [reply pkg/combustion/elemental.go:29]: I may have been confusing myself with error messages. I'm happy to revert.

File: docs/building-images.md [hunk 5]
```diff
@@ -144,14 +144,16 @@ the structure of this directory will be better fleshed out. For now, the require
 
 There are a number of optional directories that may be included in the image configuration directory:
 
-* `network` - If present, network configurations will be generated from all desired states in this directory
-  and will be included in the built image. The configurations relevant for the particular host will be identified
-  and applied during the combustion phase. Check [nm-configurator](https://github.com/suse-edge/nm-configurator/)
-  for more information.
+* `certificates` - If present, all files with the extension ".pem" will be installed as CA certificates
+  in the built image. Note that ".crt" files are permitted, but please ensure they have a ".pem" extension.
```
Comment (atanasdinov) [inline docs/building-images.md:148]: The second sentence is no longer applicable.

**Review by atanasdinov — APPROVED** [review]

**Review by rdoxenham — APPROVED** [review]
LGTM 🚀

**Pull request merged**